framework/utils/mathUtils.py

```diff
@@ -253,6 +253,14 @@ def calculateStats(data):
   ret["kurtosis"] = stats.kurtosis(data)
   return ret
 
+def hyperdiagonal(lengths):
+  """
+    Obtains the length of a diagonal of a hyperrectangle given the lengths of the sides.  Useful for high-dimensional distance scaling.
+    @ In, lengths, list(float), lengths of the sides of the ND rectangle
+    @ Out, diag, float, the length of the diagonal between furthest-separated corners of the hypercube
+  """
+  return np.sqrt(sum(x*x for x in lengths))
```

Comment:
I see my mistake here.  This is true when the number of entries is very large; however, when the number of dimensions is small, it's a bit faster to use the "sum" method.  However, the more problematic cases will be when we have large dimension, so as you say it makes more sense to use the numpy method.  Fixed.
```
>>> timeit.timeit("np.sqrt(np.sum(lengths**2))",setup="import numpy as np;lengths=np.random.rand(2)",number=1)
3.886222839355469e-05
>>>timeit.timeit("np.sqrt(sum(lengths**2))",setup="import numpy as np;lengths=np.random.rand(2)",number=1)
2.5987625122070312e-05
```